pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -122,21 +126,110 @@ func haveOverlap(a1, a2 []string) bool {
 
 func (pl *VolumeRestrictions) PreFilter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod) *framework.Status {
 	if pl.enableReadWriteOncePod {
-		return pl.isReadWriteOncePodAccessModeConflict(pod)
+		return pl.isReadWriteOncePodAccessModeConflict(cycleState, pod)
 	}
+	p := &preFilterState{}
+	cycleState.Write(preFilterStateKey, p)
 	return framework.NewStatus(framework.Success)
 }
 
 // isReadWriteOncePodAccessModeConflict checks if a pod uses a PVC with the ReadWriteOncePod access mode.
 // This access mode restricts volume access to a single pod on a single node. Since only a single pod can
 // use a ReadWriteOncePod PVC, mark any other pods attempting to use this PVC as UnschedulableAndUnresolvable.
-// TODO(#103132): Mark pod as Unschedulable and add preemption logic.
-func (pl *VolumeRestrictions) isReadWriteOncePodAccessModeConflict(pod *v1.Pod) *framework.Status {
+func (pl *VolumeRestrictions) isReadWriteOncePodAccessModeConflict(cycleState *framework.CycleState, pod *v1.Pod) *framework.Status {
 	nodeInfos, err := pl.nodeInfoLister.NodeInfos().List()
 	if err != nil {
 		return framework.NewStatus(framework.Error, "error while getting node info")
 	}
+	pvcKeys, status := pl.keysForPVCWithRWOP(pod)
+	if status != nil {
+		return status
+	}
+	var nodesMatchPVCWithRWOP []*v1.Node
+	ctx, cancel := context.WithCancel(context.Background())
+	processNode := func(i int) {
```

Comment:
nit: I think it might be worth adding a comment explaining what's happening here. From my understanding, we're collecting the nodes that contain all of the pod's PVCs